libbeat/common/kubernetes/metadata/resource.go

```diff
@@ -118,7 +118,9 @@ func (r *Resource) GenerateK8s(kind string, obj kubernetes.Resource, options ...
 			case "Deployment",
 				"ReplicaSet",
 				"StatefulSet",
-				"DaemonSet":
+				"DaemonSet",
+				"Job",
```

Comment:
with this change we also can close this PR: https://github.com/elastic/beats/pull/28954

Reply:
@tetianakravchenko my PR is a little bit tricky cause has some tricky parts around k8s api versions. I think we can move on with the community PR and I can rebase mine on top of it. wdyt?